src/NuGet.Core/NuGet.Protocol.Core.v3/CachingSourceProvider.cs

```diff
@@ -21,8 +21,8 @@ public class CachingSourceProvider : ISourceRepositoryProvider
         private readonly List<SourceRepository> _repositories = new List<SourceRepository>();
 
         // There should only be one instance of the source repository for each package source.
-        private readonly ConcurrentDictionary<PackageSource, SourceRepository> _cachedSources
-            = new ConcurrentDictionary<PackageSource, SourceRepository>();
+        private readonly ConcurrentDictionary<string, SourceRepository> _cachedSources
+            = new ConcurrentDictionary<string, SourceRepository>(StringComparer.OrdinalIgnoreCase);
```

Comment:
It should be consistent with URL case sensitivity. Consider reverting this change here or adopting "ignore case" above.